### DISCUSSION TITLE
I've got this (  `RadioGroupItem` must be used within `RadioGroup` ) error in my codebase when I create a separate component.

### DISCUSSION THREAD

**Discussion by sumon-chandra** [shadcn-ui/ui#3597] [Q&A] [answered]
I'm building a multi-step form with ShadcnUI. I've used form components along with the checkbox, select and radio group components. Everything is working fine when I write all logic in a single component ( this is the component - [https://github.dev/sumon-chandra/multi-step-form/blob/main/src/components/form/form2.tsx](https://github.dev/sumon-chandra/multi-step-form/blob/main/src/components/form/form2.tsx) ). 

But, when I create components for each step of the form I've got an error on the RadioGroup component. It says `` `RadioGroupItem` must be used within `RadioGroup`` `. 
👇👇
![image](https://github.com/shadcn-ui/ui/assets/80484412/857e83d1-6a24-489a-9b3c-fa69085950fd)

This is the components for each step -- 👇👇

![image](https://github.com/shadcn-ui/ui/assets/80484412/c95dc78b-2b68-48cc-8cc9-78d4a45fd280)

And this is the RadioGroup ``FormField`` component. 👇👇

![image](https://github.com/shadcn-ui/ui/assets/80484412/2fb3039b-ea6e-4175-97a2-93d675afd50d)

### I don't understand why I'm getting this error since my ``RadioGroupItem`` is in the ``RadioGroup`` component.

**Comment by joaopedrodcf** [comment, marked as answer]
Can you check that the `RadioGroupItem` and the `RadioGroupItem` that you are importing are not coming from radix instead of your local abstraction with shadcn ? Sometimes auto imports do that to me

**Reply by sumon-chandra** [reply]
## It works. 
I was importing the ``RadioGroup`` from radix by mistake. 🤦‍♂️
